src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -69,18 +69,24 @@ public SaveDatabaseAction(BasePanel panel, JabRefPreferences prefs, BibEntryType
     }
 
     private boolean saveDatabase(Path file, boolean selectedOnly, Charset encoding, SavePreferences.DatabaseSaveType saveType) throws SaveException {
-        try {
-            SavePreferences preferences = prefs.loadForSaveFromPreferences()
-                                               .withEncoding(encoding)
-                                               .withSaveType(saveType);
-
-            AtomicFileWriter fileWriter = new AtomicFileWriter(file, preferences.getEncoding(), preferences.makeBackup());
+        SavePreferences preferences = prefs.loadForSaveFromPreferences()
+                                           .withEncoding(encoding)
+                                           .withSaveType(saveType);
+        try (AtomicFileWriter fileWriter = new AtomicFileWriter(file, preferences.getEncoding(), preferences.makeBackup())) {
             BibtexDatabaseWriter databaseWriter = new BibtexDatabaseWriter(fileWriter, preferences, entryTypesManager);
 
             if (selectedOnly) {
                 databaseWriter.savePartOfDatabase(panel.getBibDatabaseContext(), panel.getSelectedEntries());
+                // each written entry is not changed anymore
+                panel.getSelectedEntries().forEach(bibEntry -> {
+                    bibEntry.setChanged(false);
```

Comment:
I think then you also need to reset the parsed serialization. Otherwise the following code
https://github.com/JabRef/jabref/blob/9c7f7f61c40443b3688af16c0d95c54a78900f87/src/main/java/org/jabref/logic/bibtex/BibEntryWriter.java#L60-L63
results in a wrong serialization for the following use case:
- Open library
- Change entry
- Save (entry written correctly)
- Save again (entry is back to original entry)

I'm actually not sure if it is worth to change the semantics of `changed` (right now it is "changed since library was loaded" and you now want it to be "changed since library was last saved").

Reply:
Thank you so much for the hint!

This prepares my fix for the file-changed-on-disk issue and will enable other features. Therefore, I need a flag indicating the change from the last save.

Think, to change the semantics to the desired one leads to a duplication of the writing code. Thus, I've introduced a `saved` flag in my new commit. -- And added a test covered your concern.

Reply:
I don't think the current solution works either. The BibEntryWriter is also used for backups / temporary files.

Can you explain in a bit more detail what you want to do in the big picture. Maybe we could find a proper fix together.